processor/transformprocessor/internal/metrics/processor.go

```diff
@@ -41,64 +43,64 @@ func NewProcessor(statements []string, functions map[string]interface{}, setting
 	for i := 0; i < td.ResourceMetrics().Len(); i++ {
 		rmetrics := td.ResourceMetrics().At(i)
-		ctx.resource = rmetrics.Resource()
+		ctx.Resource = rmetrics.Resource()
 		for j := 0; j < rmetrics.ScopeMetrics().Len(); j++ {
 			smetrics := rmetrics.ScopeMetrics().At(j)
-			ctx.il = smetrics.Scope()
+			ctx.InstrumentationScope = smetrics.Scope()
 			metrics := smetrics.Metrics()
-			ctx.metrics = metrics
+			ctx.Metrics = metrics
 			for k := 0; k < metrics.Len(); k++ {
-				ctx.metric = metrics.At(k)
-				switch ctx.metric.DataType() {
+				ctx.Metric = metrics.At(k)
+				switch ctx.Metric.DataType() {
 				case pmetric.MetricDataTypeSum:
-					p.handleNumberDataPoints(ctx, ctx.metric.Sum().DataPoints())
+					p.handleNumberDataPoints(ctx, ctx.Metric.Sum().DataPoints())
 				case pmetric.MetricDataTypeGauge:
-					p.handleNumberDataPoints(ctx, ctx.metric.Gauge().DataPoints())
+					p.handleNumberDataPoints(ctx, ctx.Metric.Gauge().DataPoints())
 				case pmetric.MetricDataTypeHistogram:
-					p.handleHistogramDataPoints(ctx, ctx.metric.Histogram().DataPoints())
+					p.handleHistogramDataPoints(ctx, ctx.Metric.Histogram().DataPoints())
 				case pmetric.MetricDataTypeExponentialHistogram:
-					p.handleExponetialHistogramDataPoints(ctx, ctx.metric.ExponentialHistogram().DataPoints())
+					p.handleExponetialHistogramDataPoints(ctx, ctx.Metric.ExponentialHistogram().DataPoints())
 				case pmetric.MetricDataTypeSummary:
-					p.handleSummaryDataPoints(ctx, ctx.metric.Summary().DataPoints())
+					p.handleSummaryDataPoints(ctx, ctx.Metric.Summary().DataPoints())
 				}
 			}
 		}
 	}
 	return td, nil
 }
 
-func (p *Processor) handleNumberDataPoints(ctx metricTransformContext, dps pmetric.NumberDataPointSlice) {
+func (p *Processor) handleNumberDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.NumberDataPointSlice) {
 	for i := 0; i < dps.Len(); i++ {
-		ctx.dataPoint = dps.At(i)
+		ctx.DataPoint = dps.At(i)
 		p.callFunctions(ctx)
 	}
 }
 
-func (p *Processor) handleHistogramDataPoints(ctx metricTransformContext, dps pmetric.HistogramDataPointSlice) {
+func (p *Processor) handleHistogramDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.HistogramDataPointSlice) {
 	for i := 0; i < dps.Len(); i++ {
-		ctx.dataPoint = dps.At(i)
+		ctx.DataPoint = dps.At(i)
 		p.callFunctions(ctx)
 	}
 }
 
-func (p *Processor) handleExponetialHistogramDataPoints(ctx metricTransformContext, dps pmetric.ExponentialHistogramDataPointSlice) {
+func (p *Processor) handleExponetialHistogramDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.ExponentialHistogramDataPointSlice) {
 	for i := 0; i < dps.Len(); i++ {
-		ctx.dataPoint = dps.At(i)
+		ctx.DataPoint = dps.At(i)
 		p.callFunctions(ctx)
 	}
 }
 
-func (p *Processor) handleSummaryDataPoints(ctx metricTransformContext, dps pmetric.SummaryDataPointSlice) {
+func (p *Processor) handleSummaryDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.SummaryDataPointSlice) {
```

Comment:
`tqlmetrics.MetricTransformContext` can be renamed to `tqlmetrics.TransformContext`

Reply:
Separate PR, we can look into these kind of name improvements.

Reply:
Should probably be consistent and do that style of naming for all 3 contexts.   I'll do that tomorrow.

Reply:
@bogdandrutu any issue with the context's structs names exactly matching the name of the [tql interface](TransformContext) they are meeting, and also each other?  I know it is all different packages, but might be confusing

Reply:
Here is what it looks like: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12895